File: src/types/supplier.ts
export interface Supplier {
  id: string;
  name: string;
  contactPerson?: string;
  phone: string;
  email?: string;
  address?: string;
  gstin?: string;
  paymentTerms?: "Immediate" | "Net 7" | "Net 15" | "Net 30";
  notes?: string;
  createdAt: string;
  isActive: boolean;
}

File: pos-backend/src/main/java/com/retailflow/pos/supplier/Supplier.java
package com.retailflow.pos.supplier;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.UUID;
import lombok.*;

@Data @Builder @NoArgsConstructor @AllArgsConstructor
@Entity @Table(name="suppliers")
public class Supplier {
  public enum PaymentTerms { IMMEDIATE, NET_7, NET_15, NET_30 }
  @Id private String id;
  @Column(nullable=false) private String name;
  private String contactPerson;
  private String phone;
  private String email;
  @Lob private String address;
  private String gstin;
  @Enumerated(EnumType.STRING) private PaymentTerms paymentTerms;
  @Lob private String notes;
  private boolean active;
  private LocalDateTime createdAt;
  @PrePersist void create(){ if(id==null) id=UUID.randomUUID().toString(); createdAt=LocalDateTime.now(); active=true; }
}
